### ISSUE TITLE
Node can't boot because mountpoint of all io queues is "undefined"

### ISSUE THREAD

**Issue by denesb** [scylladb/scylla#3718]
*Installation details*
Scylla version (or git commit hash): master(4a274ee7e226382a3686fe037de00a0c71faf566)
Cluster size: 3
OS (RHEL/CentOS/Ubuntu/AWS AMI): CentOS7 (GCE)

There are two distinct backtraces:
```
[Backtrace #0]
{/tmp/scylla} 0x0000000000694e02: boost::condition_variable::~condition_variable() at /opt/scylladb/include/boost/thread/pthread/condition_variable_fwd.hpp:105
{/tmp/scylla} 0x00000000005ac78c: seastar::promise<>::abandoned() at /home/botond/after/seastar/core/future.hh:1263
{/tmp/scylla}  (inlined by) seastar::promise<>::~promise() at /home/botond/after/seastar/core/future.hh:459
{/tmp/scylla}  (inlined by) seastar::stream<seastar::directory_entry>::~stream() at /home/botond/after/seastar/core/stream.hh:134
{/tmp/scylla}  (inlined by) ~work at /home/botond/after/seastar/core/reactor.cc:2348
{/tmp/scylla}  (inlined by) ~shared_ptr_no_esft at /home/botond/after/seastar/core/shared_ptr.hh:157
{/tmp/scylla}  (inlined by) dispose at /home/botond/after/seastar/core/shared_ptr.hh:213
{/tmp/scylla}  (inlined by) ~lw_shared_ptr at /home/botond/after/seastar/core/shared_ptr.hh:300
{/tmp/scylla}  (inlined by) ~<lambda> at /home/botond/after/seastar/core/reactor.cc:2423
{/tmp/scylla}  (inlined by) ~<lambda> at /home/botond/after/seastar/core/future.hh:955
{/tmp/scylla}  (inlined by) ~continuation at /home/botond/after/seastar/core/future.hh:410
{/tmp/scylla} 0x00000000005aca35: seastar::future_state<>::get_exception() const & at /home/botond/after/seastar/core/future.hh:381
{/tmp/scylla}  (inlined by) seastar::promise<>::abandoned() at /home/botond/after/seastar/core/future.hh:1266
{/tmp/scylla}  (inlined by) seastar::promise<>::~promise() at /home/botond/after/seastar/core/future.hh:459
{/tmp/scylla}  (inlined by) ~<lambda> at /home/botond/after/seastar/core/future.hh:955
{/tmp/scylla}  (inlined by) ~continuation at /home/botond/after/seastar/core/future.hh:410
{/tmp/scylla} 0x00000000005acae3: seastar::future_state<>::~future_state() at /home/botond/after/seastar/core/future.hh:318
{/tmp/scylla}  (inlined by) seastar::promise<>::~promise() at /home/botond/after/seastar/core/future.hh:460
{/tmp/scylla}  (inlined by) seastar::stream<seastar::directory_entry>::~stream() at /home/botond/after/seastar/core/stream.hh:134
{/tmp/scylla}  (inlined by) ~work at /home/botond/after/seastar/core/reactor.cc:2348
{/tmp/scylla}  (inlined by) ~shared_ptr_no_esft at /home/botond/after/seastar/core/shared_ptr.hh:157
{/tmp/scylla}  (inlined by) dispose at /home/botond/after/seastar/core/shared_ptr.hh:213
{/tmp/scylla}  (inlined by) ~lw_shared_ptr at /home/botond/after/seastar/core/shared_ptr.hh:300
{/tmp/scylla}  (inlined by) ~<lambda> at /home/botond/after/seastar/core/reactor.cc:2373
{/tmp/scylla}  (inlined by) ~<lambda> at /home/botond/after/seastar/core/future.hh:955
{/tmp/scylla}  (inlined by) ~continuation at /home/botond/after/seastar/core/future.hh:410
{/lib64/libpthread.so.0} 0x000000000000f6cf: _L_unlock_13 at funlockfile.c:?
{/lib64/libc.so.6} 0x0000000000036276: __GI_raise at :?
{/lib64/libc.so.6} 0x0000000000037967: __GI_abort at :?
{/opt/scylladb/lib64/libstdc++.so.6} 0x000000000009253c: __gnu_cxx::__verbose_terminate_handler() at /usr/src/debug/gcc-7.3.1-20180125/obj-x86_64-redhat-linux/x86_64-redhat-linux/libstdc++-v3/libsupc++/../../../../libstdc++-v3/libsupc++/vterminate.cc:50
{/opt/scylladb/lib64/libstdc++.so.6} 0x0000000000090165: __cxxabiv1::__terminate(void (*)()) at /usr/src/debug/gcc-7.3.1-20180125/obj-x86_64-redhat-linux/x86_64-redhat-linux/libstdc++-v3/libsupc++/../../../../libstdc++-v3/libsupc++/eh_terminate.cc:47
{/opt/scylladb/lib64/libstdc++.so.6} 0x000000000008f068: __cxa_call_terminate at /usr/src/debug/gcc-7.3.1-20180125/obj-x86_64-redhat-linux/x86_64-redhat-linux/libstdc++-v3/libsupc++/../../../../libstdc++-v3/libsupc++/eh_call.cc:54
{/opt/scylladb/lib64/libstdc++.so.6} 0x000000000008fa57: __gxx_personality_v0 at /usr/src/debug/gcc-7.3.1-20180125/obj-x86_64-redhat-linux/x86_64-redhat-linux/libstdc++-v3/libsupc++/../../../../libstdc++-v3/libsupc++/eh_personality.cc:676
{/opt/scylladb/lib64/libgcc_s.so.1} 0x000000000000ff42: _Unwind_GetTextRelBase at /usr/src/debug/gcc-7.3.1-20180125/obj-x86_64-redhat-linux/x86_64-redhat-linux/libgcc/../../../libgcc/unwind.inc:62
{/opt/scylladb/lib64/libgcc_s.so.1} 0x000000000001076d: _Unwind_Resume at /usr/src/debug/gcc-7.3.1-20180125/obj-x86_64-redhat-linux/x86_64-redhat-linux/libgcc/../../../libgcc/unwind.inc:230
{/tmp/scylla} 0x00000000006617ba: std::_Function_base::~_Function_base() at /opt/scylladb/include/c++/7/bits/std_function.h:276
{/tmp/scylla}  (inlined by) std::function<bool ()>::~function() at /opt/scylladb/include/c++/7/bits/std_function.h:389
{/tmp/scylla}  (inlined by) seastar::reactor::run() at /home/botond/after/seastar/core/reactor.cc:3221
{/tmp/scylla} 0x000000000074355d: _ZN7seastar7promiseIJEE8scheduleIZNS_6futureIJEE12then_wrappedIZNS4_12finally_bodyIZNS_19file_data_sink_impl5closeEvEUlvE_Lb1EEclEOS4_EUlT_E_S4_EET0_OSB_EUlSE_E_EEvSE_ at /home/botond/after/seastar/core/future.hh:551
{/tmp/scylla}  (inlined by) _ZN7seastar6futureIJEE8scheduleIZNS1_12then_wrappedIZNS1_12finally_bodyIZNS_19file_data_sink_impl5closeEvEUlvE_Lb1EEclEOS1_EUlT_E_S1_EET0_OS9_EUlSC_E_EEvSC_ at /home/botond/after/seastar/core/future.hh:762
{/tmp/scylla}  (inlined by) _ZN7seastar6futureIJEE12then_wrappedIZNS1_12finally_bodyIZNS_19file_data_sink_impl5closeEvEUlvE_Lb1EEclEOS1_EUlT_E_S1_EET0_OS8_ at /home/botond/after/seastar/core/future.hh:999
{/tmp/scylla}  (inlined by) seastar::future<>::finally_body<seastar::file_data_sink_impl::close()::{lambda()#1}, true>::operator()(seastar::future<>&&) at /home/botond/after/seastar/core/future.hh:1078
{/tmp/scylla}  (inlined by) seastar::future<> seastar::futurize<seastar::future<> >::apply<seastar::future<>::finally_body<seastar::file_data_sink_impl::close()::{lambda()#1}, true>, seastar::future<> >(seastar::future<>::finally_body<seastar::file_data_sink_impl::close()::{lambda()#1}, true>&&, seastar::future<>&&) at /home/botond/after/seastar/core/future.hh:1395
{/tmp/scylla}  (inlined by) seastar::future<> seastar::future<>::then_wrapped<seastar::future<>::finally_body<seastar::file_data_sink_impl::close()::{lambda()#1}, true>, seastar::future<> >(seastar::future<>::finally_body<seastar::file_data_sink_impl::close()::{lambda()#1}, true>&&) at /home/botond/after/seastar/core/future.hh:993
{/tmp/scylla}  (inlined by) seastar::future<> seastar::future<>::finally<seastar::file_data_sink_impl::close()::{lambda()#1}>(seastar::file_data_sink_impl::close()::{lambda()#1}&&) at /home/botond/after/seastar/core/future.hh:1051
{/tmp/scylla}  (inlined by) seastar::file_data_sink_impl::close() at /home/botond/after/seastar/core/fstream.cc:428
{/lib64/libpthread.so.0} 0x0000000000007e24: start_thread at pthread_create.c:?
{/lib64/libc.so.6} 0x00000000000febac: __clone at ??:?
```

```
[Backtrace #1]
{/tmp/scylla} 0x0000000000694e02: boost::condition_variable::~condition_variable() at /opt/scylladb/include/boost/thread/pthread/condition_variable_fwd.hpp:105
{/tmp/scylla} 0x00000000005ac78c: seastar::promise<>::abandoned() at /home/botond/after/seastar/core/future.hh:1263
{/tmp/scylla}  (inlined by) seastar::promise<>::~promise() at /home/botond/after/seastar/core/future.hh:459
{/tmp/scylla}  (inlined by) seastar::stream<seastar::directory_entry>::~stream() at /home/botond/after/seastar/core/stream.hh:134
{/tmp/scylla}  (inlined by) ~work at /home/botond/after/seastar/core/reactor.cc:2348
{/tmp/scylla}  (inlined by) ~shared_ptr_no_esft at /home/botond/after/seastar/core/shared_ptr.hh:157
{/tmp/scylla}  (inlined by) dispose at /home/botond/after/seastar/core/shared_ptr.hh:213
{/tmp/scylla}  (inlined by) ~lw_shared_ptr at /home/botond/after/seastar/core/shared_ptr.hh:300
{/tmp/scylla}  (inlined by) ~<lambda> at /home/botond/after/seastar/core/reactor.cc:2423
{/tmp/scylla}  (inlined by) ~<lambda> at /home/botond/after/seastar/core/future.hh:955
{/tmp/scylla}  (inlined by) ~continuation at /home/botond/after/seastar/core/future.hh:410
{/tmp/scylla} 0x00000000005aca35: seastar::future_state<>::get_exception() const & at /home/botond/after/seastar/core/future.hh:381
{/tmp/scylla}  (inlined by) seastar::promise<>::abandoned() at /home/botond/after/seastar/core/future.hh:1266
{/tmp/scylla}  (inlined by) seastar::promise<>::~promise() at /home/botond/after/seastar/core/future.hh:459
{/tmp/scylla}  (inlined by) ~<lambda> at /home/botond/after/seastar/core/future.hh:955
{/tmp/scylla}  (inlined by) ~continuation at /home/botond/after/seastar/core/future.hh:410
{/tmp/scylla} 0x00000000005acae3: seastar::future_state<>::~future_state() at /home/botond/after/seastar/core/future.hh:318
{/tmp/scylla}  (inlined by) seastar::promise<>::~promise() at /home/botond/after/seastar/core/future.hh:460
{/tmp/scylla}  (inlined by) seastar::stream<seastar::directory_entry>::~stream() at /home/botond/after/seastar/core/stream.hh:134
{/tmp/scylla}  (inlined by) ~work at /home/botond/after/seastar/core/reactor.cc:2348
{/tmp/scylla}  (inlined by) ~shared_ptr_no_esft at /home/botond/after/seastar/core/shared_ptr.hh:157
{/tmp/scylla}  (inlined by) dispose at /home/botond/after/seastar/core/shared_ptr.hh:213
{/tmp/scylla}  (inlined by) ~lw_shared_ptr at /home/botond/after/seastar/core/shared_ptr.hh:300
{/tmp/scylla}  (inlined by) ~<lambda> at /home/botond/after/seastar/core/reactor.cc:2373
{/tmp/scylla}  (inlined by) ~<lambda> at /home/botond/after/seastar/core/future.hh:955
{/tmp/scylla}  (inlined by) ~continuation at /home/botond/after/seastar/core/future.hh:410
{/lib64/libpthread.so.0} 0x000000000000f6cf: _L_unlock_13 at funlockfile.c:?
{/lib64/libc.so.6} 0x0000000000036276: __GI_raise at :?
{/lib64/libc.so.6} 0x0000000000037967: __GI_abort at :?
{/opt/scylladb/lib64/libstdc++.so.6} 0x0000000000092574: __gnu_cxx::__verbose_terminate_handler() at /usr/src/debug/gcc-7.3.1-20180125/obj-x86_64-redhat-linux/x86_64-redhat-linux/libstdc++-v3/libsupc++/../../../../libstdc++-v3/libsupc++/vterminate.cc:95
{/opt/scylladb/lib64/libstdc++.so.6} 0x0000000000090165: __cxxabiv1::__terminate(void (*)()) at /usr/src/debug/gcc-7.3.1-20180125/obj-x86_64-redhat-linux/x86_64-redhat-linux/libstdc++-v3/libsupc++/../../../../libstdc++-v3/libsupc++/eh_terminate.cc:47
{/opt/scylladb/lib64/libstdc++.so.6} 0x000000000008f068: __cxa_call_terminate at /usr/src/debug/gcc-7.3.1-20180125/obj-x86_64-redhat-linux/x86_64-redhat-linux/libstdc++-v3/libsupc++/../../../../libstdc++-v3/libsupc++/eh_call.cc:54
{/opt/scylladb/lib64/libstdc++.so.6} 0x000000000008fa57: __gxx_personality_v0 at /usr/src/debug/gcc-7.3.1-20180125/obj-x86_64-redhat-linux/x86_64-redhat-linux/libstdc++-v3/libsupc++/../../../../libstdc++-v3/libsupc++/eh_personality.cc:676
{/opt/scylladb/lib64/libgcc_s.so.1} 0x000000000000ff42: _Unwind_GetTextRelBase at /usr/src/debug/gcc-7.3.1-20180125/obj-x86_64-redhat-linux/x86_64-redhat-linux/libgcc/../../../libgcc/unwind.inc:62
{/opt/scylladb/lib64/libgcc_s.so.1} 0x000000000001076d: _Unwind_Resume at /usr/src/debug/gcc-7.3.1-20180125/obj-x86_64-redhat-linux/x86_64-redhat-linux/libgcc/../../../libgcc/unwind.inc:230
{/tmp/scylla} 0x00000000006617ba: std::_Function_base::~_Function_base() at /opt/scylladb/include/c++/7/bits/std_function.h:276
{/tmp/scylla}  (inlined by) std::function<bool ()>::~function() at /opt/scylladb/include/c++/7/bits/std_function.h:389
{/tmp/scylla}  (inlined by) seastar::reactor::run() at /home/botond/after/seastar/core/reactor.cc:3221
{/tmp/scylla} 0x000000000074355d: _ZN7seastar7promiseIJEE8scheduleIZNS_6futureIJEE12then_wrappedIZNS4_12finally_bodyIZNS_19file_data_sink_impl5closeEvEUlvE_Lb1EEclEOS4_EUlT_E_S4_EET0_OSB_EUlSE_E_EEvSE_ at /home/botond/after/seastar/core/future.hh:551
{/tmp/scylla}  (inlined by) _ZN7seastar6futureIJEE8scheduleIZNS1_12then_wrappedIZNS1_12finally_bodyIZNS_19file_data_sink_impl5closeEvEUlvE_Lb1EEclEOS1_EUlT_E_S1_EET0_OS9_EUlSC_E_EEvSC_ at /home/botond/after/seastar/core/future.hh:762
{/tmp/scylla}  (inlined by) _ZN7seastar6futureIJEE12then_wrappedIZNS1_12finally_bodyIZNS_19file_data_sink_impl5closeEvEUlvE_Lb1EEclEOS1_EUlT_E_S1_EET0_OS8_ at /home/botond/after/seastar/core/future.hh:999
{/tmp/scylla}  (inlined by) seastar::future<>::finally_body<seastar::file_data_sink_impl::close()::{lambda()#1}, true>::operator()(seastar::future<>&&) at /home/botond/after/seastar/core/future.hh:1078
{/tmp/scylla}  (inlined by) seastar::future<> seastar::futurize<seastar::future<> >::apply<seastar::future<>::finally_body<seastar::file_data_sink_impl::close()::{lambda()#1}, true>, seastar::future<> >(seastar::future<>::finally_body<seastar::file_data_sink_impl::close()::{lambda()#1}, true>&&, seastar::future<>&&) at /home/botond/after/seastar/core/future.hh:1395
{/tmp/scylla}  (inlined by) seastar::future<> seastar::future<>::then_wrapped<seastar::future<>::finally_body<seastar::file_data_sink_impl::close()::{lambda()#1}, true>, seastar::future<> >(seastar::future<>::finally_body<seastar::file_data_sink_impl::close()::{lambda()#1}, true>&&) at /home/botond/after/seastar/core/future.hh:993
{/tmp/scylla}  (inlined by) seastar::future<> seastar::future<>::finally<seastar::file_data_sink_impl::close()::{lambda()#1}>(seastar::file_data_sink_impl::close()::{lambda()#1}&&) at /home/botond/after/seastar/core/future.hh:1051
{/tmp/scylla}  (inlined by) seastar::file_data_sink_impl::close() at /home/botond/after/seastar/core/fstream.cc:428
{/lib64/libpthread.so.0} 0x0000000000007e24: start_thread at pthread_create.c:?
{/lib64/libc.so.6} 0x00000000000febac: __clone at ??:?
```

As reason for the crash the following reasons appear in the logs:
```
aug 22 10:07:36 botond-benchmark-node1 scylla[5214]: terminate called after throwing an instance of 'FATAL: Exception during startup, aborting: std::runtime_error (registering metrics twice for metrics: reactor_io_queue_requests)
aug 22 10:07:36 botond-benchmark-node1 scylla[5214]: std::runtime_error'
aug 22 10:07:36 botond-benchmark-node1 scylla[5214]: what():  registering metrics twice for metrics: reactor_io_queue_requests
```
```
aug 22 10:09:40 botond-benchmark-node1 scylla[5301]: scylla: /opt/scylladb/include/boost/thread/pthread/condition_variable_fwd.hpp:102: boost::condition_variable::~condition_variable(): Assertion `!ret' failed.
```
```
aug 22 09:55:07 botond-benchmark-node1 scylla[4662]: scylla: /home/botond/after/seastar/util/spinlock.hh:84: seastar::util::spinlock::~spinlock(): Assertion `!_busy.load(std::memory_order_relaxed)' failed.
```

See the complete logs: [journal.txt](https://github.com/scylladb/scylla/files/2310244/journal.txt)

I also have a very unhelpful core, with only a fraction of the frames (due to the heavy inlining). I'll upload it if can be useful still.

**Comment by avikivity** [comment]
Try starting scylla with gdb and `catch throw`, to get information about the error at the point it occurs. I've seen it before but don't recall anything else.

**Comment by denesb** [comment]
I've seen such complaints about metrics when an io_priority_class was used cross cores. I'll try what you suggested, the complication is that I can't reproduce this locally, only on GCE, when used through `systemd`. Will try to hack gdb in there somehow.

**Comment by amnonh** [comment]
For that error to happen, we should have multiple io queues with the same mountpoint, @vladzcloudius can you take a look at reactor.cc (around line 2612)

**Comment by denesb** [comment]
@amnonh as far as I can tell I have a single mountpoint:

/etc/scylla.d/io_properties.yaml
```
disks:
  - mountpoint: /var/lib/scylla
    read_iops: 360530
    read_bandwidth: 1357480064
    write_iops: 200520
    write_bandwidth: 816519104
```

**Comment by denesb** [comment]
I just read your comment again, you are talking about multiple io queues for the same mountpoint, not multiple mountpoints. :man_facepalming:

**Comment by amnonh** [comment]
Exactly

**Comment by avikivity** [comment]
Try using the same command line used by systemd.

**Comment by denesb** [comment]
The command line systemd uses is:
```
/tmp/scylla --log-to-syslog 1 --log-to-stdout 0 --default-log-level info --network-stack posix --num-io-queues=7 --io-properties-file=/etc/scylla.d/io_properties.yaml --cpuset 0-15
```

**Comment by denesb** [comment]
Can this be a configuration error? Or a scylla bug? Or both?
To give a little context, I set up my GCE nodes when benchmarking as follows:
* Install scylla nigthly
* Run `scylla_setup`
* Clone scylla.git with desired branch
* Compile scylla.
* `cp build/release/scylla /tmp`
* Patch `./lib/systemd/system/scylla-server.service` so that it starts `/tmp/scylla` instead of `/usr/bin/scylla`
* Profit: enjoy a standard setup managed with systemd, running a custom executable

**Comment by denesb** [comment]
Can this cause incompatibilities because of running a scylla nightly environment with a possibly more recent executable?

**Comment by amnonh** [comment]
It's all go down to the same issue, you probably have 3 io queue one that actually has the mountpoint and two others that are undefined.
Are all those queues you are defining used?

**Comment by denesb** [comment]
What do you mean by "used"? I didn't define any io queues, I arrived to the current configuration by running `scylla_setup`.

**Comment by denesb** [comment]
The two io queues of this reactor:
```
$10 = {_priority_classes = std::unordered_map with 0 elements, _fq = {_config = {capacity = 128, tau = {__r = 100000}, max_req_count = 4294967295, max_bytes_count = 4294967295}, _requests_executing = 0, _req_count_executing = 0,
    _bytes_count_executing = 0, _requests_queued = 0, _base = {__d = {__r = 17571599680693}}, _handles = std::priority_queue wrapping: std::vector of length 0, capacity 0, _all_classes = std::unordered_set with 0 elements}, static _max_classes = 2048,
  static _registered_shares = {_M_elems = {{<std::__atomic_base<unsigned int>> = {static _S_alignment = 4, _M_i = 0}, static is_always_lock_free = true} <repeats 2048 times>}}, static _registered_names = {_M_elems = {"" <repeats 2048 times>}},
  static read_request_base_count = 128, _config = {coordinator = 4, io_topology = std::vector of length 16, capacity 16 = {0, 0, 2, 2, 4, 4, 6, 6, 8, 8, 10, 10, 12, 12, 0, 2}, capacity = 613566756, max_req_count = 4294967295,
    max_bytes_count = 4294967295, disk_req_write_to_read_multiplier = 1, disk_bytes_write_to_read_multiplier = 1, mountpoint = "undefined"}}
```

```
$11 = {_priority_classes = std::unordered_map with 0 elements, _fq = {_config = {capacity = 128, tau = {__r = 100000}, max_req_count = 4294967295, max_bytes_count = 4294967295}, _requests_executing = 0, _req_count_executing = 0,
    _bytes_count_executing = 0, _requests_queued = 0, _base = {__d = {__r = 17571599681735}}, _handles = std::priority_queue wrapping: std::vector of length 0, capacity 0, _all_classes = std::unordered_set with 0 elements}, static _max_classes = 2048,
  static _registered_shares = {_M_elems = {{<std::__atomic_base<unsigned int>> = {static _S_alignment = 4, _M_i = 0}, static is_always_lock_free = true} <repeats 2048 times>}}, static _registered_names = {_M_elems = {"" <repeats 2048 times>}},
  static read_request_base_count = 128, _config = {coordinator = 4, io_topology = std::vector of length 16, capacity 16 = {0, 0, 2, 2, 4, 4, 6, 6, 8, 8, 10, 10, 12, 12, 0, 2}, capacity = 613566756, max_req_count = 4294967295,
    max_bytes_count = 4294967295, disk_req_write_to_read_multiplier = 1, disk_bytes_write_to_read_multiplier = 1, mountpoint = "undefined"}}
```

This is shard 4:
```
(gdb) p _id
$12 = 4
```

**Comment by denesb** [comment]
As mentioned above, there are two `io_queue`s, one for the [placeholder mountpoint](https://github.com/scylladb/seastar/blob/9bb1611bb2e9e856fc497aaa68f32ac509b8065c/core/reactor.cc#L4032-L4033) and one for the configured mounpoint, coming from the `io-properties` file.
The problem is that both have "undefined" as mounpoint name in `_config::mounpoint`. "undefined" is the default value for the `mountpoint` field so this hints at this field not receiving a value after the struct is created. Indeed, in [generate_config()](https://github.com/scylladb/seastar/blob/9bb1611bb2e9e856fc497aaa68f32ac509b8065c/core/reactor.cc#L4061-L4083) there is a branch and the `mountpoint` member is only set on the true branch.

**Comment by denesb** [comment]
@avikivity is it intended for io queues to have `mountpoint` set to undefined under some circumstances?

**Comment by denesb** [comment]
The branch was introduced by this commit: https://github.com/scylladb/seastar/commit/d5ad5f84dede3a5a8abdaba4184c191b947d91d0
@glommer can you comment?

**Comment by glommer** [comment]
Patch on the list (seastar-dev)

It's essentially a pretty stupid mistake I made in the last patch. Sorry about that

**Comment by denesb** [comment]
Fixed by https://github.com/scylladb/seastar/commit/12f18ce06056978355f74f48c07c0286762f6523. Will close when scylla's seastar is updated to contain that commit.

**Comment by avikivity** [comment]
Fixed by 22396d57c228262bb85b72e64973a7620615d531.
